go-controller/pkg/libovsdb/util/northd_sync.go

```diff
@@ -42,24 +42,27 @@ func WaitUntilNorthdSyncOnce(ctx context.Context, nbClient, sbClient client.Clie
 	if err != nil {
 		return fmt.Errorf("failed to generate ops to mutate nb_cfg: %w", err)
 	}
-	expectedNbCfgValue := nbGlobal.NbCfg + 1
+	expectedHvCfgValue := nbGlobal.NbCfg + 1
 	if _, err = libovsdbops.TransactAndCheck(nbClient, ops); err != nil {
 		return fmt.Errorf("failed to transact to increment nb_cfg: %w", err)
 	}
-	sbGlobal := &sbdb.SBGlobal{}
-	// poll until we see the expected value in SB DB every 5 milliseconds until context is cancelled.
+	nbGlobal = &nbdb.NBGlobal{}
+	// ovn-northd sets hv_cfg to the smallest sequence number of all the chassis in the system,
+	// as reported in the Chassis_Private table in the southbound database. Thus, hv_cfg
+	// equals nb_cfg if all chassis are caught up with NB DB.
+	// poll until we see the expected value in NB DB every 5 milliseconds until context is cancelled.
 	err = wait.PollUntilContextCancel(ctx, time.Millisecond*5, true, func(_ context.Context) (done bool, err error) {
-		if sbGlobal, err = libovsdbops.GetSBGlobal(sbClient, sbGlobal); err != nil {
+		if nbGlobal, err = libovsdbops.GetNBGlobal(nbClient, nbGlobal); err != nil {
 			// northd hasn't added an entry yet
 			if errors.Is(err, client.ErrNotFound) {
 				return false, nil
 			}
-			return false, fmt.Errorf("failed to get sb_global table entry from SB DB: %w", err)
+			return false, fmt.Errorf("failed to get nb_global table entry from NB DB: %w", err)
 		}
-		return sbGlobal.NbCfg >= expectedNbCfgValue, nil // we only need to ensure it is greater than or equal to the expected value
+		return nbGlobal.HvCfg >= expectedHvCfgValue, nil // we only need to ensure it is greater than or equal to the expected value
```

Comment:
And I forgot to add also here, during the wrap being it negative or 0 as mentioned above, this will always be true. So it will return prematurely. For that we do extra check in form of `nbGlobal.HvCfg - expectedHvCfgValue < int32.maxValue`, so the whole condition would be `return nbGlobal.HvCfg >= expectedHvCfgValue && nbGlobal.HvCfg - expectedHvCfgValue < int32.maxValue, nil`.

Reply:
I see the value to 0 if its negative so we can avoid this.